backend/lib/deal-submitter.js

```diff
@@ -0,0 +1,115 @@
+/** @import {PgPool, Queryable} from '@filecoin-station/deal-observer-db' */
+import Cursor from 'pg-cursor'
+
+/**
+ * Find deals that are eligible to be submitted to a spark api and submits them.
+ *
+ * @param {PgPool} pgPool
+ * @param {string} sparkApiBaseURL
+ * @param {string} dealIngestionAccessToken
+ * @param {number} batchSize
+ * @param {(sparkApiBaseURL: string, dealIngestionAccessToken: string, eligibleDeals: Array) => Promise<void>} submitEligibleDealsFn
+ */
+export const findAndSubmitEligibleDeals = async (pgPool, sparkApiBaseURL, dealIngestionAccessToken, batchSize, submitEligibleDealsFn) => {
+  for await (const eligibleDeals of findEligibleDeals(pgPool, batchSize)) {
+    const formattedEligibleDeals = formatEligibleDeals(eligibleDeals)
+    await submitEligibleDealsFn(sparkApiBaseURL, dealIngestionAccessToken, formattedEligibleDeals)
+    await markEligibleDealsSubmitted(pgPool, eligibleDeals)
+  }
+}
+
+/**
+ * Find deals that are eligible to be submitted to a spark api.
+ * Eligible deals are those that have not been submitted yet,
+ * were created more than 2 days ago, have payload cid, and have not yet expired.
```

Comment:
```suggestion
 * Eligible deals are those that
 * - have not been submitted yet
 * - were created more than 2 days ago
 * - have payload cid
 * - have not yet expired
```

Reply:
I think you're right on this. Maybe we should drop "eligible" altogether and just go with "unsubmitted", "unposted" or "unsynced" so we would end up with functions name like `findUnsubmittedDeals`, `findUnpostedDeals` or `findUnsyncedDeals`.